src/Transport/Receiving/MessagePump.cs

```diff
@@ -272,28 +272,30 @@ await processor.CloseAsync(cancellationToken)
             // args.CancellationToken is currently not used because the v8 version that supports cancellation was designed
             // to not flip the cancellation token until the very last moment in time when the stop token is flipped.
             var contextBag = new ContextBag();
+            using var processingTokenSource = CancellationTokenSource.CreateLinkedTokenSource(messageProcessingCancellationToken);
+
+            processMessageEventArgs.MessageLockLostAsync += MessageLockLostHandler;
 
             try
             {
-                using (var azureServiceBusTransaction = CreateTransaction(message.PartitionKey))
-                {
-                    contextBag.Set(message);
-                    contextBag.Set(processMessageEventArgs);
+                using var azureServiceBusTransaction = CreateTransaction(message.PartitionKey);
+                contextBag.Set(message);
+                contextBag.Set(processMessageEventArgs);
 
-                    var messageContext = new MessageContext(messageId, headers, body, azureServiceBusTransaction.TransportTransaction, ReceiveAddress, contextBag);
+                var messageContext = new MessageContext(messageId, headers, body,
+                    azureServiceBusTransaction.TransportTransaction, ReceiveAddress, contextBag);
 
-                    await onMessage(messageContext, messageProcessingCancellationToken).ConfigureAwait(false);
+                await onMessage(messageContext, processingTokenSource.Token).ConfigureAwait(false);
```

Comment:
it may make sense to elevate this into its own `var`
`var messageProcessingCancellationToken = processingTokenSource.Token;`

Reply:
The problem is that this name is already taken by the pump token. I will rename it to `lockLostCancellationToken`